accepted/future-releases/unquoted-imports/feature-specification.md

```diff
@@ -353,11 +353,11 @@ parse while looking like a single opaque "unquoted string" to users and tools.
 
 The normative stuff starts now. Here is the proposal:
 
-We add a new rule and hang it off the existing `uri` rule already used by import
-and export directives:
+We add a new rule and hang it off the existing `configurableUri` rule already
+used by import and export directives:
 
 ```
-uri               ::= stringLiteral | packagePath
+configurableUri   ::= ( stringLiteral | packagePath ) configurationUri*
```

Comment:
Would also need to update `configurationUri` to allow both quoted URIs and package-paths.
Current grammar is:
```ebnf
<uri> ::= <stringLiteral>
<configurableUri> ::= <uri> <configurationUri>*
<configurationUri> ::= \IF{} `(' <uriTest> `)' <uri>  ;; <-- second occurrence of `uri`.
```

(We really should have a parameterized grammar:
```dart
<configurableRef>[ref] ::=  ref <configurationRef>[ref]*
<configurationRef>[ref] ::= 'if' '(' <uriTest> ')' ref
```
so you could write:
```dart
<importDirective> ::= <metadata> 'import' <configurableRef>[<stringLiteral>|<packagePath>] ...
<partDirective> ::= <metadata> 'part' <configurableRef>[<stringLiteral>] ';'
```
Would avoid a lot of parallel duplication in the grammar.)

Reply:
Good catch. I took a slightly different approach and put `packagePath` inside `uri` so that `configurableUri` and `configurationUri` can be left alone (with a note about how parts-with-imports affects it).

And, yes, parametric grammar rules would be *so* nice.